File: pact-jvm-provider-junit/src/main/java/au/com/dius/pact/provider/junit/Consumer.java
package au.com.dius.pact.provider.junit;

import java.lang.annotation.ElementType;
import java.lang.annotation.Inherited;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * Used to pass consumer name to {@link PactRunner}
 */
@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.TYPE)
@Inherited
public @interface Consumer {
    /**
     * @return consumer name for pact test running
     */
    String value();
}

File: provider/src/main/kotlin/au/com/dius/pact/provider/junitsupport/Consumer.kt
package au.com.dius.pact.provider.junitsupport

import java.lang.annotation.Inherited
import kotlin.annotation.Retention

/**
 * Used to pass consumer name to Pact runner
 */
@Retention(AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.CLASS, AnnotationTarget.FILE)
@Inherited
annotation class Consumer(
  /**
   * @return consumer name for pact test running
   */
  val value: String
)
